core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -20,146 +20,146 @@
-        throw Error(
-          'Trying to add an input before the response for the previous ' +
-          'input has been received.',
-          transcript);
-      }
-      if (!isHint) {
-        numAnswersSubmitted += 1;
-      }
-      transcript[transcript.length - 1].inputResponsePairs.push({
-        learnerInput: input,
-        oppiaResponse: null,
-        isHint: isHint
-      });
-    },
-    addNewResponse: function(response) {
-      var pairs = transcript[transcript.length - 1].inputResponsePairs;
-      if (pairs[pairs.length - 1].oppiaResponse !== null) {
-        throw Error(
-          'Trying to add a response when it has already been added.',
-          transcript);
-      }
-      pairs[pairs.length - 1].oppiaResponse = response;
-    },
-    getNumCards: function() {
-      return transcript.length;
-    },
-    getCard: function(index) {
-      if (index < 0 || index >= transcript.length) {
-        $log.error(
-          'Requested card with index ' + index +
-          ', but transcript only has length ' + transcript.length + ' cards.');
-      }
-      return transcript[index];
-    },
-    getLastAnswerOnActiveCard: function(index) {
-      if (this.isLastCard(index) || transcript[index].stateName === null) {
-        return null;
-      } else {
-        return transcript[index].
-          inputResponsePairs.slice(-1)[0].learnerInput;
+        lastCard.setDestStateName(newDestStateName);
+      },
+      addNewInput: function(input, isHint) {
+        var pairs = transcript[transcript.length - 1].getInputResponsePairs();
+        if (pairs.length > 0 &&
+            pairs[pairs.length - 1].oppiaResponse === null) {
+          throw Error(
+            'Trying to add an input before the response for the previous ' +
+            'input has been received.',
+            transcript);
+        }
+        if (!isHint) {
+          numAnswersSubmitted += 1;
+        }
+        transcript[transcript.length - 1].addInputResponsePair({
+          learnerInput: input,
+          oppiaResponse: null,
+          isHint: isHint
+        });
+      },
+      addNewResponse: function(response) {
+        var pairs = transcript[transcript.length - 1].getInputResponsePairs();
+        if (pairs[pairs.length - 1].oppiaResponse !== null) {
+          throw Error(
+            'Trying to add a response when it has already been added.',
+            transcript);
+        }
+        pairs[pairs.length - 1].oppiaResponse = response;
+      },
+      getNumCards: function() {
+        return transcript.length;
+      },
+      getCard: function(index) {
+        if (index < 0 || index >= transcript.length) {
+          $log.error(
+            'Requested card with index ' + index +
+            ', but transcript only has length ' +
+            transcript.length + ' cards.');
+        }
+        return transcript[index];
+      },
+      getLastAnswerOnActiveCard: function(index) {
```

Comment:
Where is the "active" card stored?

Maybe rename arg to activeCardIndex.

Reply:
Done.